[AG-10043] Feature(tooltip): allow setting headerTooltipValueGetter

sdwvit opened the pull request:
Feature(tooltip): add headerTooltipValueGetter for customizable tooltips 
- Introduced headerTooltipValueGetter to allow dynamic tooltip content.
 - Updated colDef, colDefValidations, gridOptions, and properties.json.
 - Enhanced tooltipService to prioritize headerTooltipValueGetter over headerTooltip.

local plnkr: https://plnkr.co/edit/riwX9Rc6ecKCH9d5?open=main.js
same but with column groups: https://plnkr.co/edit/NPMeY8YQqrX18I1c?open=main.js

<img width="1346" height="1532" alt="Screenshot_20251013_163649" src="https://github.com/user-attachments/assets/7b461ff6-e455-4dfc-b4d3-67f313a3a9f4" />


Fix [AG-10043](https://ag-grid.atlassian.net/browse/AG-10043)

Commit:
Feature(tooltip): add headerTooltipValueGetter for customizable tooltips

 - Introduced headerTooltipValueGetter to allow dynamic tooltip content.
 - Updated colDef, colDefValidations, gridOptions, and properties.json.
 - Enhanced tooltipService to prioritize headerTooltipValueGetter over headerTooltip.

Signed-off-by: Victor Musienko <victor.m@noibu.com>

Commit:
- Simplified getTooltipValue logic using optional chaining and nullish coalescing
 - Updated references from colGroupDef to colDef for consistency
 - Improved readability of tooltip value retrieval in both header and headerGroup

Signed-off-by: Victor Musienko <victor.m@noibu.com>

Commit:
update example and docs

Signed-off-by: Victor Musienko <victor.m@noibu.com>

Commit:
finish sentence

Signed-off-by: Victor Musienko <victor.m@noibu.com>

StephenCooper commented on packages/ag-grid-community/src/tooltip/tooltipService.ts:
```diff
@@ -56,14 +57,10 @@ export class TooltipService extends BeanStub implements NamedBean {
-            },
+            getTooltipValue: () =>
+                value ??
+                colDef?.headerTooltipValueGetter?.(_addGridCommonParams(gos, { location: 'cell', colDef, column })) ??
```
Is 'cell' the correct choice here? Also we should consider if we are going to set the value / valueFormatted params.

StephenCooper reviewed (changes requested):
I think we need to check the params object being passed to the callback.

sdwvit committed:
- Updated tooltipService to support customizable tooltip values using passed parameters.
 - Improved logic for determining tooltip display names based on column definitions.

sdwvit committed:
lint

StephenCooper commented on .run/Header Group Tooltip Value Getter feat.run.xml:
Are these run files accidentally committed?

sdwvit committed:
update per pr comment

sdwvit committed:
remove unused ColDef import from tooltipService
